### PR TITLE
Spec request

### PR DESCRIPTION
**Pull request by ChaabaneAnas**
- Add RSpec as a gem to your project.
- Create a Request spec file for all your controllers.
- Make sure that for each action you check:
    - If response status was correct.
    - If a correct template was rendered.
    - If the response body includes correct placeholder text.

### Need a big picture?

### CODE REVIEW

File: spec/requests/posts_spec.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    end
+  end
+
+  describe 'GET /index' do
+    it 'returns http success' do
+      get '/users/1/posts/'
+      expect(response).to have_http_status(:success)
+    end
+  end
+end
```
Comment (Nemwel-Boniface) [inline spec/requests/posts_spec.rb]: - [ ] Good job working on your controller specs. However, you seem to only have handled one end of the equation. Kindly include tests that ensures correct view template is rendered. Example : expect(response).to render_template(:show) 
- [x] Kindly also include tests that ensures the response body includes the correct placeholder text. Example : expect(response.body).to include("Welcome to user index page")

File: spec/requests/users_spec.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    end
+  end
+
+  describe 'GET /show' do
+    it 'returns http success' do
+      get '/users/1'
+      expect(response).to have_http_status(:success)
+    end
+  end
+end
```
Comment (Nemwel-Boniface) [inline spec/requests/users_spec.rb:17]: - [ ] Good job working on your controller specs. However, you seem to only have handled one end of the equation. Kindly include tests that ensures correct view template is rendered. Example : expect(response).to render_template(:show) 
- [x] Kindly also include tests that ensures the response body includes the correct placeholder text. Example : expect(response.body).to include("Welcome to user index page")

**Review by Nemwel-Boniface — CHANGES_REQUESTED** [review]
## Hello @ChaabaneAnas    :wave: , 
I'm Nemwel your reviewer for this PR :man_beard: 

Good job working on your project so far! :clap: 
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights

Great job setting up the correct linters ✔️ 
Professional documentation ✔️ 
Correct workflow ✔️ 
Meaningful commit messages ✔️  


## ♻️ Required Changes ♻️

_Check the comments under the review._

## [Optional] suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding! :partying_face: 👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me **@Nemwel-Boniface** in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/requests/users_spec.rb (hunk 2 above)
Comment (ChaabaneAnas) [reply spec/requests/users_spec.rb:17]: @Nemwel-Boniface Hello mate yes, 
i did forget to implement this requirement:
- If the response body includes correct placeholder text.

but the template test is already done within the: `spec/view` directory: 
image

File: spec/views/users/show.html.erb_spec.rb [hunk 3]
```diff
@@ -0,0 +1,8 @@
+require 'rails_helper'
+
+RSpec.describe 'users/show.html.erb', type: :view do
```
Comment (Ismailco) [inline spec/views/users/show.html.erb_spec.rb:3]: - [ ] Spec file should include tests that ensure the correct view template is rendered, as the previous reviewer mentioned kindly consider including those tests
e.g  `expect(response).to render_template(:show)`

File: spec/views/users/index.html.erb_spec.rb [hunk 4]
```diff
@@ -0,0 +1,8 @@
+require 'rails_helper'
+
+RSpec.describe 'users/index.html.erb', type: :view do
```
Comment (Ismailco) [inline spec/views/users/index.html.erb_spec.rb:3]: - [ ] Spec file should include tests that ensure the correct view template is rendered, as the previous reviewer mentioned kindly consider including those tests
e.g  `expect(response).to render_template(:show)`

File: spec/views/posts/show.html.erb_spec.rb [hunk 5]
```diff
@@ -0,0 +1,8 @@
+require 'rails_helper'
+
+RSpec.describe 'posts/show.html.erb', type: :view do
```
Comment (Ismailco) [inline spec/views/posts/show.html.erb_spec.rb:3]: - [ ] Spec file should include tests that ensure the correct view template is rendered, as the previous reviewer mentioned kindly consider including those tests
e.g  `expect(response).to render_template(:show)`

File: spec/views/posts/index.html.erb_spec.rb [hunk 6]
```diff
@@ -0,0 +1,8 @@
+require 'rails_helper'
+
+RSpec.describe 'posts/index.html.erb', type: :view do
```
Comment (Ismailco) [inline spec/views/posts/index.html.erb_spec.rb:3]: - [ ] Spec file should include tests that ensure the correct view template is rendered, as the previous reviewer mentioned kindly consider including those tests
e.g  `expect(response).to render_template(:show)`

**Review by Ismailco — CHANGES_REQUESTED** [review]
### Hi @ChaabaneAnas, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't been addressed yet.


 ## Highlights

✔️  No linter errors
✔️ Used Gitflow
✔️ Work documented professionally

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by DeliceLydia — APPROVED** [review]
## Hi @ChaabaneAnas, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
# Congratulations! 🎉 



### Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**